examples/allocation_benchmark.c

```diff
@@ -5,7 +5,8 @@
 
 int main() {
     volatile uint64_t * tmp;
-    mmtk_gc_init(1024*1024*1024);
+    mmtk_set_heap_size(1024*1024*1024);
+    mmtk_gc_init();
```

Comment:
The `mmtk_set_heap_size` in the example may give the reader a false impression that the heap size can be changed during execution, which is not true.  Without a concrete implementation as in dummyvm, the reader may not realise that it sets an option in the builder before creating an MMTk instance.

For the purpose of example, I think it is OK to keep the old code `mmtk_gc_init(1024 * 1024 * 1024);` because the C API doesn't have to correspond one-to-one with what `memory_manager` provides.  Conceptually, a C function `mmtk_gc_init` could both set the option and create the instance.  We can also change the name to `mmtk_init` or `init_mmtk` to avoid confusion.

Reply:
Done. I changed it to `mmtk_init(heap_size)` (`gc_init()` is renamed to `mmtk_init()` -- I think it is a good idea, as it avoids the confusion with `initialize_collection()`). As for the C API, in this PR, I moved `examples/mmtk.h` to `docs/header/mmtk.h`. This is meant to be the canonical header, and the C signatures in the header tries to be one-to-one mapped from the Rust API. The DummyVM binding uses a header in `vmbindings/dummyvm/api/mmtk.h`, which is different from the canonical version.